### PR TITLE
vcpu_hotpluggable:add case for cold plug unplug

### PR DESCRIPTION
**Pull request by dzhengfy**
Case ID: RHEL-112474
    Test scenario:
    - Cold plug/unplug vcpu with shutoff vm, check xml, restart libvirt, check domain

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
libvirt-7.4.0-1.el9.x86_64

```
#  avocado run --vt-type libvirt --vt-machine-type q35 vcpu_hotpluggable
JOB ID     : 0e5a86448df27356bb68ce5e34bb277e0b2b2d4b
JOB LOG    : /root/avocado/job-results/job-2021-06-25T04.52-0e5a864/job.log
 (01/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.vcpus_with_order: PASS (44.28 s)
 (02/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.enable: FAIL: cpu number in VM is not correct, it should be 5 cpus (39.90 s)
 (03/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.disable: PASS (40.40 s)
 (04/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.unplug.live: PASS (40.33 s)
 (05/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.unplug.config.running_vm: PASS (40.24 s)
 (06/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.unplug.config.shutoff_vm: PASS (5.12 s)
 (07/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.plug.live: FAIL: cpu number in VM is not correct, it should be 5 cpus (40.35 s)
 (08/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.plug.config.running_vm: PASS (40.62 s)
 (09/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.positive_test.plug.config.shutoff_vm: PASS (5.20 s)
 (10/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.negative_test.dup_order1: PASS (5.68 s)
 (11/11) type_specific.io-github-autotest-libvirt.vcpu_hotpluggable.negative_test.dup_order2: PASS (5.71 s)
RESULTS    : PASS 9 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 308.57 s
```

2 failures are due to existing bugs 1903243

File: libvirt/tests/src/cpu/vcpu_hotpluggable.py [hunk 1]
```diff
@@ -18,6 +18,43 @@
...
+    else:
+        test.error("No vcpu is enabled")
+    # Check  <vcpu id='x' enabled='xx' ... order='x'/> should disappear
+    vcpu_order_list = re.findall(r"vcpu.*order=.*", re_dump_xml)
+    if vcpu_order_list:
+        test.fail("vcpu order info should be cleared, "
+                  "but found {}".format(vcpu_order_list))
+
+
+def check_vm_exist(test, vm_name, state):
```
Comment (chunfuwen) [inline libvirt/tests/src/cpu/vcpu_hotpluggable.py:46]: it sounds like we have virsh.domain_exists(vm_name, uri=uri):
 to help check whether VM exist or not

Comment (dzhengfy) [reply libvirt/tests/src/cpu/vcpu_hotpluggable.py:46]: @chunfuwen , I need to check the given vm exists with given vm state. But virsh.domain_exists() does not handle vm state. So it does not meet my needs exactly. 
Several functions can be used to meet my needs, like virsh.domstate() or virsh.dom_list().

**Review by kamilvarga — APPROVED** [review]
LGTM.

**Review by Yingshun — APPROVED** [review]

**Review by chunfuwen — APPROVED** [review]
LGTM

**Pull request merged**